File: Array/238ProductofArrayExceptSelf.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> rev(nums.size(),1);
        int p = 1;
        for(int i=0; i<nums.size(); i++){
            rev[i] = p;
            p *= nums[i];
        }
        p = 1;
        for(int i=nums.size()-1; i>=0; i--){
            rev[i] *= p;
            p *= nums[i];
        }
        return rev;
    }
};

File: Array/238productofArrayExceptSelf.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = []
        p = 1
        for i in range(len(nums)):
            res.append(p)
            p *= nums[i]
        p = 1
        for i in range(len(nums)-1,-1,-1):
            res[i] *= p
            p *= nums[i]
            
        return res